--- kernel-build/mm/madvise.c.orig	2026-05-04 08:50:54.802408500 -0700
+++ kernel-build/mm/madvise.c	2026-05-04 08:50:58.023436500 -0700
@@ -1098,6 +1098,15 @@
 		break;
 	case MADV_COLLAPSE:
 		return madvise_collapse(vma, prev, start, end);
+	case MADV_SEMANTIC_STREAMING:
+		vma->vm_semantic_hint = VM_SEMANTIC_STREAMING;
+		return 0;
+	case MADV_SEMANTIC_REUSE:
+		vma->vm_semantic_hint = VM_SEMANTIC_REUSE;
+		return 0;
+	case MADV_SEMANTIC_EPHEMERAL:
+		vma->vm_semantic_hint = VM_SEMANTIC_EPHEMERAL;
+		return 0;
 	}
 
 	anon_name = anon_vma_name(vma);
