--- a/mm/madvise.c
+++ b/mm/madvise.c
@@ -1055,6 +1055,10 @@
 	case MADV_RANDOM:
 		new_flags = (new_flags & ~VM_SEQ_READ) | VM_RAND_READ;
 		break;
+	case MADV_SEMANTIC_STREAMING:
+	case MADV_SEMANTIC_REUSE:
+	case MADV_SEMANTIC_EPHEMERAL:
+		return 0;
 	case MADV_DONTFORK:
 		new_flags |= VM_DONTCOPY;
 		break;
@@ -1189,6 +1193,9 @@
 	case MADV_NOHUGEPAGE:
 	case MADV_COLLAPSE:
 #endif
+	case MADV_SEMANTIC_STREAMING:
+	case MADV_SEMANTIC_REUSE:
+	case MADV_SEMANTIC_EPHEMERAL:
 	case MADV_DONTDUMP:
 	case MADV_DODUMP:
 	case MADV_WIPEONFORK:
